problem_builder/v1/upgrade.py

```diff
@@ -148,15 +144,46 @@ def parse_xml_for_HtmlDescriptor(cls, node, runtime, keys, id_generator):
     blocks_found = []
 
     def find_mentoring_blocks(block):
-        if isinstance(block, MentoringBlock) and block.xml_content:  # If it's a v1 block it will have xml_content
+        """
+        Find mentoring and recently-upgraded blocks. We check the recently upgraded ones
+        in case an error happened and we're re-running the upgrade.
+        """
+        # If it's a v1 block or a recently upgraded block it will have xml_content:
+        if isinstance(block, (MentoringBlock, NewMentoringBlock)) and block.xml_content:
             blocks_found.append(block.scope_ids.usage_id)
         elif block.has_children:
             for child_id in block.children:
                 find_mentoring_blocks(block.runtime.get_block(child_id))
     find_mentoring_blocks(course)
+
     total = len(blocks_found)
     print(" ➔ Found {} mentoring blocks".format(total))
 
+    print(" ➔ Doing a quick sanity check of the url_names")
```

Comment:
Yeah, there are actually a lot of non-unicode strings in this file that contain unicode characters. My bad. Because they're just being treated as bytes and getting piped directly to stdout, it hasn't caused a problem, but you're right that they should be marked as Unicode.